src/interfaces/records-write.ts

```diff
@@ -663,7 +664,12 @@ export class RecordsWrite {
     if (contextId !== undefined) { authorizationPayload.contextId = contextId; } // assign `contextId` only if it is defined
     if (attestationCid !== undefined) { authorizationPayload.attestationCid = attestationCid; } // assign `attestationCid` only if it is defined
     if (encryptionCid !== undefined) { authorizationPayload.encryptionCid = encryptionCid; } // assign `encryptionCid` only if it is defined
-    if (permissionsGrantId !== undefined) { authorizationPayload.permissionsGrantId = permissionsGrantId; }
+    if (additionalProperties?.permissionsGrantId !== undefined) {
+      authorizationPayload.permissionsGrantId = additionalProperties?.permissionsGrantId;
+    }
+    if (additionalProperties?.protocolRole !== undefined) {
+      authorizationPayload.protocolRole = additionalProperties?.protocolRole;
+    }
```

Comment:
Design: signature payload include properties that are not included in the descriptor (such as grant ID), but if protocol role is included descriptor, then is doesn't need to be included again in the payload.

But it seems that we ought to not tie role information used mainly for authorization purpose to the record itself, so we probably should remove the role info from the descriptor (seen below) like how grants are handled. Maybe confirm with @csuwildcat with this thinking.

Reply:
Updated -- Totally agree it should just be in `authorization`, I had meant to clean it up from the descriptor before opening the PR.